Code/skyrim_ui/src/app/services/setting.service.ts

```diff
@@ -20,115 +20,87 @@ export const fontSizeToPixels: Record<FontSize, number> = {
+    super(inital_value)
   }
 
-  public setPartyAnchor(anchor: PartyAnchor) {
-    this.storeService.set('party_anchor', anchor);
+  public next(value: number) {
+    this.storeService.set(this.storeKey, value)
+    super.next(value)
   }
+}
 
-  public getPartyAnchor(): PartyAnchor {
-    return JSON.parse(this.storeService.get('party_anchor', PartyAnchor.TOP_LEFT));
-  }
+class ToggleSetting extends Setting<boolean> {
 
-  public setPartyAnchorOffsetX(offset: number) {
-    this.storeService.set('party_anchor_offset_x', offset);
+  constructor(readonly storeService: StoreService, private storeKey: string, defaultValue: boolean) {
+    const inital_value = storeService.getBool(storeKey, defaultValue);
+    super(inital_value)
   }
 
-  public getPartyAnchorOffsetX(): number {
-    return JSON.parse(this.storeService.get('party_anchor_offset_x', 0));
+  public next(value: boolean) {
+    this.storeService.set(this.storeKey, value)
+    super.next(value)
   }
+}
 
-  public setPartyAnchorOffsetY(offset: number) {
-    this.storeService.set('party_anchor_offset_y', offset);
-  }
+class SelectSetting extends Setting<string> {
 
-  public getPartyAnchorOffsetY(): number {
-    return JSON.parse(this.storeService.get('party_anchor_offset_y', 3));
+  constructor(
+    private readonly storeService: StoreService, 
+    private storeKey: string, 
+    private options: string[], 
+    defaultValue: string
+  ) {
+    const storedValue = storeService.get(storeKey, defaultValue);
+    super(options.includes(storedValue) ? storedValue : defaultValue);
   }
 
-  public setFontSize(size: FontSize) {
-    this.fontSizeChange.next(size);
-    this.storeService.set('font_size', size);
+  public next(value: string) {
+    if (this.options.includes(value)) {
+      this.storeService.set(this.storeKey, value);
+      super.next(value);
+    }
   }
+}
 
-  public getFontSize(): FontSize {
-    return this.storeService.get('font_size', FontSize.M);
-  }
+@Injectable({
+  providedIn: 'root',
+})
+export class SettingService {
 
-  public getLanguage(): string {
-    const language = this.storeService.get('language', 'en');
-    const isValid = !!(this.translocoService.getAvailableLangs() as any[])
-      .find(lang => lang === language || lang.id === language);
-    if (isValid) {
-      return language;
-    }
-    return this.translocoService.getDefaultLang();
+  private readonly languageValues: string[] = Object.values(this.translocoService.getAvailableLangs()).map(lang => lang.id);
+  private readonly fontSizeValues = Object.values(FontSize);
+  private readonly partyAnchorValues = Object.values(PartyAnchor);
+  private readonly autoHideTimeValues = autoHideTimerLengths;
+
+  public settings = {
+    volume: new SliderSetting(this.storeService, "audio_volume", 0.5),
+    muted: new ToggleSetting(this.storeService, "audio_muted", false),
+    language: new SelectSetting(this.storeService, "language", this.languageValues, "en"),
```

Comment:
We should probably call `this.translocoService.getDefaultLang()` to get the default, thoughts?